src/processor.rs

```diff
@@ -793,9 +801,186 @@ impl<'a> ReplayProcessor<'a> {
             self.known_demolishes.push((demolish, index))
         }
 
+        // New path: consider DemolishExtended and Demolish attribute updates directly from this frame
+        // ---- DEMOLISH DETECTION + BOOST PAD TRACKING (DUAL LOGIC, FOR CURRENT BOXCARS) ----
+        for update in frame.updated_actors.iter() {
+
+            // =========================================
+            // BOOST PAD DETECTION
+            // =========================================
+
+            // Capture boost pad positions dynamically (map-agnostic)
+            for na in frame.new_actors.iter() {
+                if let Some(name) = self.actor_state.get_object_name(na.object_id) {
+                    if name.contains("TAGame.Pickup_Boost_TA") {
+                        // Handle trajectory safely (location is Option<Vector3i>)
+                        if let Some(loc) = &na.initial_trajectory.location {
+                            // Convert Vector3i (integers) → Vector3f (floats)
+                            let locf = boxcars::Vector3f {
+                                x: loc.x as f32,
+                                y: loc.y as f32,
+                                z: loc.z as f32,
+                            };
+
+                            // Store pad position in HashMap<i32, Vector3f>
+                            self.boost_pad_positions.insert(na.actor_id.0 as i32, locf);
+
+                            // Log for debugging
+                            log::debug!(
+                                "[BOOST-PAD] Spawned pad {} at ({:.0}, {:.0}, {:.0})",
+                                na.actor_id.0,
+                                locf.x,
+                                locf.y,
+                                locf.z
+                            );
+                        }
+                    }
+                }
+            }
+
+
+
+            // =========================================
+            // BOOST PICKUP DETECTION (FINAL)
+            // =========================================
+            for update in frame.updated_actors.iter() {
```

Comment:
shouldn't this go in its own function. This function seems related to demolishes.